framework/src/dirackernels/DiracKernel.C

```diff
@@ -199,7 +201,8 @@ DiracKernel::addPoint(Point p, unsigned id)
   // If id == libMesh::invalid_uint (the default), the user is not
   // enabling caching when they add Dirac points.  So all we can do is
   // the PointLocator lookup, and call the other addPoint() method.
-  const Elem * elem = _dirac_kernel_info.findPoint(p, _mesh);
+  const Elem * elem = blockRestricted() ? _dirac_kernel_info.findPoint(p, _mesh, &blockIDs())
+                                        : _dirac_kernel_info.findPoint(p, _mesh);
```

Comment:
It would be more satisfying to me if we could modify `findPoint` to handle the value of `blockIDs()` returned when we are not `blockRestricted()` (which I believe is `ANY_BLOCK_ID`) then doing these ternary operations everywhere.